python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -542,11 +542,19 @@ def write(self, value: Any):
     def begin_read(self) -> Any:
         self.ensure_registered_as_reader()
         actor_id = ray.get_runtime_context().get_actor_id()
+        if actor_id is None:
+            # The reader is the driver process.
+            # Use the actor ID of the DAGDriverProxyActor.
+            actor_id = self._get_actor_id(self._readers[0])
```

Comment:
wrap this into a method?

Reply:
Actually, can you also assert here that len(self._readers) == 1? Want to make sure that we are picking the right reader.

Reply:
Updated https://github.com/ray-project/ray/pull/45937/commits/32d14f80166de400ca459b4c7d1b8b7c49dd9771